tests/qnn/test_qnn_torch.py

```diff
@@ -107,28 +149,26 @@ def test_weight_shape_unspecified(self, get_circuit):
             TorchLayer(c, w)
 
     @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
-    def test_var_pos(self, get_circuit, monkeypatch):
+    def test_var_pos(self):
         """Test if a TypeError is raised when instantiated with a variable number of positional
         arguments"""
-        c, w = get_circuit
-
-        class FuncPatch:
-            """Patch for variable number of keyword arguments"""
-
-            sig = c.func.sig
-            var_pos = True
-            var_keyword = False
+        dev = qml.device("default.qubit", wires=1)
+        weight_shapes = {"w1": (3, 3), "w2": 1}
 
-        with monkeypatch.context() as m:
-            m.setattr(c, "func", FuncPatch)
+        @qml.qnode(dev, interface="torch")
+        def circuit(inputs, w1, w2, *args):
+            return qml.expval(qml.PauliZ(0))
```

Comment:
Similar: makes more sense to put a user facing example here, rather than monkeypatching an existing functions' inspected signature.